azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -625,10 +626,17 @@ private void deleteExecutionDir(final int executionId) {
 
   @Override
   public void handleEvent(final Event event) {
-    if (event.getType() == EventType.FLOW_FINISHED || event.getType() == EventType.FLOW_STARTED) {
-      final FlowRunner flowRunner = (FlowRunner) event.getRunner();
-      final ExecutableFlow flow = flowRunner.getExecutableFlow();
+    // Adding extra logging for debuggability of execution directory cleanup call
+    final FlowRunner flowRunner = (FlowRunner) event.getRunner();
+    final ExecutableFlow flow = flowRunner.getExecutableFlow();
 
+    if (event.getType() != null && flow != null) {
```

Comment:
Nit: both the conditions in the `if` seem to represent an invalid state. Perhaps through an exception for it in a `else` block or at least log the occurrence of it ?

Reply:
Adding logging for the occurrence of it.